### PR TITLE
add erf op

### PR DESCRIPTION
**Pull request by iclementine**
1. add erf op;
2. add python API for erf op;
3. add unittest for erf op;
4. add document for erf op.
5. update paddle enforce.

PR for chinese documentation in fluiddoc
https://github.com/PaddlePaddle/FluidDoc/pull/1675

Screenshot_2020-01-02 erf — fluiddoc dev 文档

### CODE REVIEW

File: paddle/fluid/operators/erf_op.cu [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+#include "paddle/fluid/operators/erf_op.h"
+
+namespace ops = paddle::operators;
+REGISTER_OP_CUDA_KERNEL(
+    erf, ops::ErfKernel<paddle::platform::CUDADeviceContext, float>,
+    ops::ErfKernel<paddle::platform::CUDADeviceContext, double>);
+REGISTER_OP_CUDA_KERNEL(
```
Comment (kuke) [inline paddle/fluid/operators/erf_op.cu:24]: Please register fp16 kernel for this op and reedit docs

File: paddle/fluid/operators/erf_op.cc [hunk 2]
```diff
@@ -0,0 +1,130 @@
...
+class ErfGradOp : public framework::OperatorWithKernel {
+ public:
+  using framework::OperatorWithKernel::OperatorWithKernel;
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    PADDLE_ENFORCE(ctx->HasInput(framework::GradVarName("Out")),
+                   "Input(Out@GRAD) should not be null");
+    PADDLE_ENFORCE(ctx->HasInput("Out"), "Input(Out) should not be null");
+    auto x_grad_name = framework::GradVarName("X");
+    ctx->SetOutputDim(x_grad_name, ctx->GetInputDim("Out"));
```
Comment (kuke) [inline paddle/fluid/operators/erf_op.cc]: Also need to share lod for backward
```
ctx->ShareLoD("X", /*->*/ x_grad_name);
```

File: paddle/fluid/operators/erf_op.cc [hunk 3]
```diff
@@ -0,0 +1,130 @@
...
+
+class ErfOp : public framework::OperatorWithKernel {
+ public:
+  ErfOp(const std::string &type, const framework::VariableNameMap &inputs,
+        const framework::VariableNameMap &outputs,
+        const framework::AttributeMap &attrs)
+      : OperatorWithKernel(type, inputs, outputs, attrs) {}
+
+  void InferShape(framework::InferShapeContext *ctx) const override {
+    // platform::errors::InvalidArgument("example: %s", str)
```
Comment (kuke) [inline paddle/fluid/operators/erf_op.cc]: Remove this line

**Review by kuke — COMMENTED** [review]
Please add `test=develop` in commit message to trigger ci

File: paddle/fluid/operators/erf_op.cu (hunk 1 above)
Comment (iclementine) [reply paddle/fluid/operators/erf_op.cu:24]: Float16 support is added now, but it cannot pass unittest with fp16.

File: paddle/fluid/operators/erf_op.cc (hunk 2 above)
Comment (iclementine) [reply paddle/fluid/operators/erf_op.cc]: Added now.

File: paddle/fluid/operators/erf_op.cc (hunk 3 above)
Comment (iclementine) [reply paddle/fluid/operators/erf_op.cc]: Done.

**Review by kuke — APPROVED** [review]
LGTM

**Review by Boyan-Liu — APPROVED** [review]
LGTM

**Review by lanxianghit — APPROVED** [review]

**Pull request merged**